app/scripts/controllers/permissions/background-api.js

```diff
@@ -102,14 +100,17 @@ export function getPermissionBackgroundApiMethods({
   };
 
   const requestAccountsAndChainPermissions = async (origin, id) => {
-    // Note that we are purposely requesting an approval from the ApprovalController
-    // and then manually forming the permission that is then granted via the
-    // PermissionController rather than calling the PermissionController.requestPermissions()
-    // directly because the Approval UI is still dependent on the notion of there
-    // being separate "eth_accounts" and "endowment:permitted-chains" permissions.
-    // After that depedency is refactored, we can move to requesting "endowment:caip25"
-    // directly from the PermissionController instead.
-    const legacyApproval = await approvalController.addAndShowApprovalRequest({
+    /**
+     * Note that we are purposely requesting an approval from the ApprovalController
+     * and then manually forming the permission that is then granted via the
+     * PermissionController rather than calling the PermissionController.requestPermissions()
+     * directly because the CAIP-25 permission is missing the factory method implementation.
+     * After the factory method is added, we can move to requesting "endowment:caip25"
+     * directly from the PermissionController instead.
+     */
+    const {
+      approvedSessionScopes: { permissions },
+    } = await approvalController.addAndShowApprovalRequest({
```

Comment:
I thought part of the goal of this refactor was to get rid of this and calling `requestPermissions` directly? How/when can we do that instead then? Why not now?

Reply:
That will rely on migrating a bunch of the CAIP-25 validation/construction logic into the permission specification, i.e. creating a factory and a differ. We decided that migration could take place later and not block moving into the UI refactor and nonEVM phase.

Reply:
That's fair. Is there a ticket for this work?

Reply:
Yes.
> We have a ticket for this work on [our board](https://github.com/orgs/MetaMask/projects/146/views/1?filterQuery=label%3A%22team-wallet-api-platform%22+-status%3ABacklog&pane=issue&itemId=94981240&issue=MetaMask%7CMetaMask-planning%7C4017)

Should link you there 😅

Reply:
Perfect 👍